src/expr.cpp

```diff
@@ -4987,7 +4996,12 @@ llvm::Value *IndexExpr::GetValue(FunctionEmitContext *ctx) const {
             Assert(m->errorCount > 0);
             return nullptr;
         }
-        lvType = PointerType::GetUniform(st->GetElementType());
+        // Use varying pointer type if the index is varying, uniform otherwise
+        if (indexType->IsVaryingType()) {
+            lvType = PointerType::GetVarying(st->GetElementType());
+        } else {
+            lvType = PointerType::GetUniform(st->GetElementType());
+        }
```

Comment:
This fixes #2376, but it suggests a different expected behavior (produce an error and exit nicely), so I am confused.

Reply:
Right, the issue description was not correct, I've updated it. It's valid code and ispc should be able to compile it.